Formula/proper.rb

```diff
@@ -0,0 +1,23 @@
+class Proper < Formula
+  desc "PropEr: a QuickCheck-inspired property-based testing tool for Erlang"
+  homepage "https://proper-testing.github.io"
+  url "https://github.com/proper-testing/proper/archive/v1.3.tar.gz"
+  sha256 "7e59eeaef12c07b1e42b0891238052cd05cbead58096efdffa3413b602cd8939"
+
+  depends_on "erlang"
+
+  def install
+    system "make"
+    prefix.install Dir["ebin", "include"]
+  end
+
+  def caveats; <<~EOS
+    Add the following line to your shell startup file (~/.bashrc in the case of the Bash shell):
+      export ERL_LIBS=#{opt_prefix}
+  EOS
+  end
+
+  test do
+    assert_equal "", shell_output("find #{opt_prefix}/{ebin,include} -prune -empty -type d")
```

Comment:
We need a test that exercises the some of the functionality of the app.

Reply:
@fxcoudert This is actually a library, not an application. It doesn't have an executable to call. You need to include in your own application in order to verify it. It does come however with a test suite, which takes ~30 seconds to complete on a decent machine (i7 2.5 Ghz). Would it be acceptable to run the library test suite as part of the installation process?

Reply:
@fxcoudert Ping.

Reply:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@fxcoudert I've updated the test so that it actually tests the successful installation of the library.